File: src/main/java/no/stunor/origo/eventorapi/model/origo/event/Race.java
package no.stunor.origo.eventorapi.model.origo.event;

import java.util.Date;
import java.util.List;

import no.stunor.origo.eventorapi.model.event.DistanceEnum;
import no.stunor.origo.eventorapi.model.event.LightConditionEnum;
import no.stunor.origo.eventorapi.model.event.Position;
import no.stunor.origo.eventorapi.model.calendar.UserCompetitor;

public record Race(String raceId,
                   String name,
                   LightConditionEnum lightCondition,
                   DistanceEnum distance,
                   Date date,
                   Position position,
                   Boolean startList,
                   Boolean resultList,
                   Boolean livelox,
                   List<UserCompetitor> competitors) {
}

File: src/main/java/no/stunor/origo/eventorapi/model/event/Race.kt
package no.stunor.origo.eventorapi.model.event

import com.google.cloud.Timestamp
import java.io.Serializable

data class Race (
        var raceId: String = "",
        var name: String = "",
        var lightCondition: LightConditionEnum = LightConditionEnum.DAY,
        var distance: DistanceEnum = DistanceEnum.MIDDLE,
        var date: Timestamp? = null,
        var position: Position? = null,
        var startList: Boolean = false,
        var resultList: Boolean = false,
        var livelox: Boolean = false,
) : Serializable
